### DISCUSSION TITLE
let leaf nodes have different colors automatically

### DISCUSSION THREAD

**Discussion by neuralgitwork** [freeplane/freeplane#514] [Help]
Is it possible to achieve the following styling automatically without adjusting certain nodes manually?
<img width="422" alt="1" src="https://user-images.githubusercontent.com/66968391/181034135-fdc02b94-ffaa-4964-8f0f-26a688782f2f.PNG">
Every main branch should have different color + fitting edge color... with the addition that any node that is a leaf has altering colors.

This seems impossible to me :P

**Comment by macmarrum** [comment]
I would use Map Conditional Style and Edge Colors, similar to #424.

Please have a look at [neuralgitwork~sameColorBranchWithAlteringLeafs.zip](https://github.com/freeplane/freeplane/files/9193323/neuralgitwork.sameColorBranchWithAlteringLeafs.zip)

Here's the Script Filter contents (it's difficult to extract it from a mind map, once entered as part of a condition)

```groovy
if (!node.isRoot()) return false

import java.awt.Color
import org.freeplane.features.edge.EdgeController
import org.freeplane.features.edge.mindmapmode.MEdgeController
import org.freeplane.features.nodestyle.NodeBorderModel

final mapModel = node.mindMap.delegate
final controller = EdgeController.controller as MEdgeController

node.mindMap.root.children.eachWithIndex { level1, i ->
    def colorCounter = i + 1
    def edgeColor = controller.getEdgeColor(mapModel, colorCounter)
    Color bgColor
    level1.findAll().each { nodex ->
        NodeBorderModel.getModel(nodex.delegate).setBorderColorMatchesEdgeColor(true)
        nodex.style.edge.color = edgeColor
        if (nodex.isLeaf()) {
            if (nodex.parent.getChildPosition(nodex) % 2 == 0) {
                bgColor = edgeColor.brighter()
            } else {
                bgColor = edgeColor.darker()
            }
        } else {
            bgColor = edgeColor
        }
        nodex.style.backgroundColor = bgColor
    }
}

return true
```
